jib-core/src/test/java/com/google/cloud/tools/jib/registry/credentials/DockerCredentialHelperTest.java

```diff
@@ -95,6 +100,32 @@ public void testRetrieve()
     Mockito.verify(processBuilderFactory).apply(command);
   }
 
+  @Test
+  public void testRetrieveWithEnvironment()
+      throws CredentialHelperUnhandledServerUrlException, CredentialHelperNotFoundException,
+          IOException {
+    List<String> command = Arrays.asList(Paths.get("/foo/bar").toString(), "get");
+    Mockito.when(processBuilderFactory.apply(command)).thenReturn(processBuilder);
+    Map<String, String> processBuilderEnvironment = Mockito.spy(new HashMap<>());
+    Mockito.when(processBuilder.environment()).thenReturn(processBuilderEnvironment);
+
+    Map<String, String> credHelperEnvironment = ImmutableMap.of("ENV_VARIABLE", "Value");
+    DockerCredentialHelper credentialHelper =
+        dockerCredentialHelper(
+            "serverUrl",
+            Paths.get("/foo/bar"),
+            systemProperties,
+            processBuilderFactory,
+            credHelperEnvironment);
+    Credential credential = credentialHelper.retrieve();
+    Assert.assertEquals("myusername", credential.getUsername());
+    Assert.assertEquals("mysecret", credential.getPassword());
+
+    Mockito.verify(processBuilderFactory).apply(command);
+    Mockito.verify(processBuilderEnvironment).putAll(credHelperEnvironment);
+    Assert.assertEquals(1, processBuilderEnvironment.size());
```

Comment:
Perhaps we can check also if the environment has `ENV_VARIABLE` and its value?

Reply:
Ah, yes